src/main/java/command/format/AddDeadlineTaskCommandFormatter.java

```diff
@@ -0,0 +1,39 @@
+package com.leeyiyuan.command.format;
+
+
+import com.leeyiyuan.command.AddDeadlineTaskCommand;
+import com.leeyiyuan.command.Command;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.regex.Pattern;
+
+/** 
+ * Represents a formatter for an AddDeadlineTaskCommand. 
+ */
+public class AddDeadlineTaskCommandFormatter extends CommandFormatter {
+
+    /** 
+     * {@inheritDoc} 
+     */
+    @Override
+    public Command parse(String text) throws CommandParseException {
+        if (Pattern.matches("^deadline ?/by.*$", text)) {
```

Comment:
Now you have two problems. :(

You didn't need to use regex here, I think. Just attempt to parse and fail if something is missing.

Reply:
"Just attempt to parse and fail if something is missing" - I feel that that is the point of regex, which does in a more concise manner rather the process of having a whole chunk of code to do the checking.

Reply:
Fair enough. I felt that asserting a prior with a regex makes the parsing easier since you know that the text is valid and you can skip the messy checks in between the parsing process.